### PR TITLE
IRIS-3986 | Replace sign in code with redirects to auth pages

### PR DESCRIPTION
**Pull request by slayful**

### CODE REVIEW

File: extensions/wikia/UserLogin/UserSignupSpecialController.class.php [hunk 1]
```diff
@@ -260,445 +46,49 @@ public function signup() {
...
-		$this->result = ( $signupForm->msgType == 'error' ) ? $signupForm->msgType : 'ok';
-		$this->msg = $signupForm->msg;
-		$this->errParam = $signupForm->errParam;
-		if ( $this->request->getVal('retusername') ) {
-			$this->username = $signupForm->mUsername;
-		}
+	private function redirectToAuthPages() {
+		$out = $this->getOutput();
+		$out->redirect( $this->getRedirectUrl() );
+		$this->clearBodyAndSetMaxCache( $out );
```
Comment (slayful) [inline extensions/wikia/UserLogin/UserSignupSpecialController.class.php:69]: Pending https://github.com/Wikia/app/pull/12817 this code can be extracted to a more common place.

File: extensions/wikia/UserLogin/UserLoginSpecialController.class.php [hunk 2]
```diff
@@ -193,7 +193,7 @@ public function getCloseAccountRedirectUrl() {
 	}
 
 	public function modal() {
-		$this->redirectToMercury();
+		$this->redirectToAuthPages();
```
Comment (mszabo-wikia) [inline extensions/wikia/UserLogin/UserLoginSpecialController.class.php:196]: We could move this call to `init` method which is automatically invoked by Nirvana dispatcher before each method, so we don't have to call it everywhere.

File: extensions/wikia/UserLogin/UserLoginSpecialController.class.php [hunk 3]
```diff
@@ -55,7 +55,7 @@ private function initializeTemplate() {
 	}
 
 	private function isMonobookOrUncyclo() {
-		$skin = $this->wg->User->getSkin();
+		$skin = RequestContext::getMain()->getSkin();
```
Comment (mszabo-wikia) [inline extensions/wikia/UserLogin/UserLoginSpecialController.class.php:58]: We can use locally available `$this->getSkin()` instance here.

**Comment by jsutterfield** [comment]
LGTM

File: extensions/wikia/UserLogin/UserLoginSpecialController.class.php [hunk 4]
```diff
@@ -114,7 +114,7 @@ private function getAuthenticationResource() {
 	 * Shown for both Special:UserLogin and Special:UserSignup when visited logged in.
 	 */
 	public function loggedIn() {
-		$this->redirectToMercury();
+		$this->redirectToAuthPages();
 	}
 
 	public function loginForm() {
```
Comment (mszabo-wikia) [inline extensions/wikia/UserLogin/UserLoginSpecialController.class.php:120]: This method still contains code that handles `POST` requests. Should it be removed?

**Pull request closed without merging**